src/inmanta/server/services/compilerservice.py

```diff
@@ -272,6 +272,10 @@ async def ensure_venv() -> Optional[data.Report]:
                     return await self._end_stage(returncode=0)
 
             async def update_modules() -> data.Report:
+                # Make sure that no Inmanta packages are installed in the compiler venv
+                python_env = PythonEnvironment(env_path=venv_dir)
+                python_env.uninstall(pkg_names=PythonEnvironment.get_protected_inmanta_packages())
```

Comment:
Do I turn this into a separate stage of the compiler service or do we do it like this (in the background)?

Reply:
I hadn't considered we could just do it in the background. I think I have a slight preference for making it a separate stage, but I'm really not sure.

Reply:
I don't think so, it's still done synchronously, just hidden from the user.

Reply:
We have to log the output correctly, to the compile report, so this won't fly.